python/planout/experiment.py

```diff
@@ -315,5 +318,8 @@ def assign(self, params, **kwargs):
     def checksum(self):
         # self.script must be a dictionary
         assert hasattr(self, 'script') and type(self.script) == dict
+        src = json.dumps(self.script)
 
-        return hashlib.sha1(json.dumps(self.script)).hexdigest()[:8]
+        if not isinstance(src, six.binary_type):
+            src = src.encode("ascii")
```

Comment:
In Python 2, `str` is equal to `six.binary_type`, and `json.dumps` always returns `str` as far as I have tested:

```
>>> type(json.dumps(""))
<type 'str'>
>>> type(json.dumps(u""))
<type 'str'>
```

So the encode won't happen, thus no checksums will change.

But in Python 3, `six.binary_type` is `bytes`, `json.dumps()` will always produce `str`, but `hashlib.sha1()` only accepts `bytes`, so the encode is needed.